modules.cmake

```diff
@@ -43,25 +43,83 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif ()
 
 
+
+# Receives latest available C++ standard version
+#
+# Usage:
+#   modules_get_latest_cxx(<variable_name>)
+#   if(<variable_name> GREATER 17)
+#      ...
+#   endif()
+function(modules_get_latest_cxx result)
+  foreach(ver 26 23 20 17 14 11 98)
+    if(CMAKE_CXX_COMPILE_FEATURES MATCHES "cxx_std_${ver}")
+      set(${result} ${ver} PARENT_SCOPE)
+      break()
+    endif()
+  endforeach()
```

Comment:
I think it would be better to find all occurrences of `cxx_std_*` and return the largest instead of trying different known standard versions which would need periodic update.

Reply:
reimplemented